### PR TITLE
Optimise kselftest removing repetitive code

### PR DESCRIPTION
**Pull request by harish-24**
Patch optimizes kselftest with removal of few variables where are not needed.

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
`self.buldir` with
zip:
```
2020-07-30 05:13:21,326 parameters       L0143 DEBUG| PARAMS (key=location, path=*, default=https://github.com/torvalds/linux/archive/master.zip) => 'https://github.com/torvalds/linux/archive/master.zip'
2020-07-30 05:13:21,335 asset            L0322 INFO | Asset not in cache, fetching it.
2020-07-30 05:13:21,335 download         L0070 INFO | Fetching https://github.com/torvalds/linux/archive/master.zip -> /home/harish/tests/data/cache/by_location/53a2ed218f53fdc4c23494f8b6ef7d5464535cd3/kselftest.zip.pa2j519v
2020-07-30 05:13:22,179 download         L0054 DEBUG| Retrieved URL "https://github.com/torvalds/linux/archive/master.zip": content-length UNKNOWN, date: "Thu, 30 Jul 2020 09:09:14 GMT", last-modified: "UNKNOWN"
2020-07-30 05:14:30,392 output           L0660 DEBUG| [stdout] /var/tmp/avocado_6rwn03cb/avocado_job_4qk7exyt/1-kernel_kselftest.py_kselftest.test/linux-master
```
tar:
```
2020-07-30 05:18:30,742 parameters       L0143 DEBUG| PARAMS (key=location, path=*, default=https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.7.11.tar.xz) => 'https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.7.11.tar.xz'
2020-07-30 05:18:30,743 asset            L0322 INFO | Asset not in cache, fetching it.
2020-07-30 05:18:30,743 download         L0070 INFO | Fetching https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.7.11.tar.xz -> /home/harish/tests/data/cache/by_location/3bb9e8e2f99e9a01e781560e8fa4a1de7a47fa90/kselftest.tar.a1rxohnq
2020-07-30 05:18:31,112 download         L0054 DEBUG| Retrieved URL "https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.7.11.tar.xz": content-length 112721564, date: "Thu, 30 Jul 2020 09:14:23 GMT", last-modified: "Wed, 29 Jul 2020 08:27:39 GMT"
2020-07-30 05:19:14,970 output           L0660 DEBUG| [stdout] /var/tmp/avocado_947ihsns/avocado_job_xo0mewbx/1-kernel_kselftest.py_kselftest.test/linux-5.7.11
```

**Comment by harish-24** [comment]
> is this really a eliminating variables i did not feel, seems me just having diffrent way

Do you agree with this different way less variables are used?

**Comment by PraveenPenguin** [comment]
> > is this really a eliminating variables i did not feel, seems me just having diffrent way
> 
> Do you agree with this different way less variables are used?

no, as earlier we have `kselfdir`, `git_location` and now it is `match` ,  `x` so variable point of view  I don't feel it is optimization

**Comment by harish-24** [comment]
> > > is this really a eliminating variables i did not feel, seems me just having diffrent way
> > 
> > 
> > Do you agree with this different way less variables are used?
> 
> no, as earlier we have `kselfdir`, `git_location` and now it is `match` , `x` so variable point of view I don't feel it is optimization

Agreed. So lesser code repetition? If you agree to that I can change the commit message.

File: kernel/kselftest.py [hunk 1]
```diff
@@ -81,17 +81,10 @@ def setUp(self):
...
-            if ".zip" in location:
-                kselfdir = "kselftest.zip"
-                git_location = False
-            elif ".tar" in location:
-                kselfdir = "kselftest.tar"
-                git_location = False
-            if not git_location:
-                tarball = self.fetch_asset(kselfdir, locations=[location],
-                                           expire='1d')
+            match = next((x for x in [".zip", ".tar"] if x in location), None)
```
Comment (PraveenPenguin) [inline kernel/kselftest.py:84]: and why for `x`  travis not a complaint  about pylint standard  let me check that  and it should be meaningful

**Comment by PraveenPenguin** [comment]
> > > > is this really a eliminating variables i did not feel, seems me just having diffrent way
> > > 
> > > 
> > > Do you agree with this different way less variables are used?
> > 
> > 
> > no, as earlier we have `kselfdir`, `git_location` and now it is `match` , `x` so variable point of view I don't feel it is optimization
> 
> Agreed. So lesser code repetition? If you agree to that I can change the commit message.

yes please even for  variable name for `x` also have some meaning name having pylint standard

**Pull request merged**